aiida/orm/nodes/data/code.py

```diff
@@ -120,32 +120,51 @@ def full_label(self):
         """
         return '{}@{}'.format(self.label, self.get_computer_name())
 
+    @property
+    def label(self):
+        """Return the node label.
+
+        :return: the label
+        """
+        return super().label
+
+    @label.setter
+    def label(self, value):
+        """Set the label.
+
+        :param value: the new value to set
+        """
+        if '@' in str(value):
+            msg = "Code labels must not contain the '@' symbol"
+            raise InputValidationError(msg)
```

Comment:
This should have been just a `ValuError`. The `InputValidationError` is intended for use in `CalcJob.prepare_for_submission`

Reply:
Sorry I didn't see this - this code is just copy-pasted from where it was before.
There are a lot of places in aiida-core where this error is used that have nothing to do with prepare_for_submission:
https://github.com/aiidateam/aiida-core/blob/09cb6cb70df024143ac5a3720e4034b609536b02/aiida/orm/querybuilder.py#L141

Do all of these uses have to be replaced?

Reply:
I would say so yes, but that would be a breaking change. The `InputValidationError` is not a subclass of `ValueError` which is what one would expect for these kinds of errors. I will open an issue to address this for `2.0`